### ISSUE TITLE
contao:user:list bugs with JSON format

### ISSUE THREAD

**Issue by ausi** [contao/contao#4937]
### Affected version(s)

4.13.5

### Description

Running `contao:user:list --format=json` results in an error: 
`Malformed UTF-8 characters, possibly incorrectly encoded` 
(but this error is swallowed because we’re not using `JSON_THROW_ON_ERROR` and the output is just empty)

Running `contao:user:list --format=json --column=username --admins` results in the following output:

```
 ! [NOTE] Only showing admin accounts

[{"username":"admin"}]
```

which is invalid JSON.

IMO we should:
- Use `JSON_THROW_ON_ERROR`
- Base64 encode columns containing binary data, or skip them
- Make sure all output is JSON if `--format=JSON` is used (including errors, warnings and notices)

/cc @richardhj

**Comment by richardhj** [comment]
> Base64 encode columns containing binary data, or skip them

None of the default columns ('username', 'name', 'admin', 'dateAdded', 'lastLogin') contains binary data, so skipping is no option if one explicitly asks for it.

+1 for the agenda otherwise :-)

**Comment by ausi** [comment]
> > Base64 encode columns containing binary data, or skip them
> 
> None of the default columns ('username', 'name', 'admin', 'dateAdded', 'lastLogin') contains binary data, so skipping is no option if one explicitly asks for it.

But with `--format=json` the default columns are not used: https://github.com/contao/contao/blob/cda9669b8df8faaa53779ee141390f1ccafc8983/core-bundle/src/Command/UserListCommand.php#L134-L135

**Comment by richardhj** [comment]
Considering the use cases for the CLI, default columns is something we can introduce. ATM also superfluous information like the password hashes are exported.

**Comment by ausi** [comment]
@richardhj can you maybe provide a pull request to fix this? ☺️

**Comment by richardhj** [comment]
Addressed all issues in #5520.
